services/tables/src/main/java/com/linkedin/openhouse/tables/services/TablesServiceImpl.java

```diff
@@ -199,6 +199,44 @@ public void deleteTable(String databaseId, String tableId, String actingPrincipa
     openHouseInternalRepository.deleteById(tableDtoPrimaryKey);
   }
 
+  @Override
+  public void renameTable(
+      String fromDatabaseId,
+      String fromTableId,
+      String toDatabaseId,
+      String toTableId,
+      String tableCreatorUpdater) {
+    Optional<TableDto> tableDto =
+        openHouseInternalRepository.findById(
+            TableDtoPrimaryKey.builder().databaseId(fromDatabaseId).tableId(fromTableId).build());
+
+    if (!tableDto.isPresent()) {
+      throw new NoSuchUserTableException(fromDatabaseId, fromTableId);
+    }
+
+    Optional<TableDto> targetedTableDto =
+        openHouseInternalRepository.findById(
+            TableDtoPrimaryKey.builder().databaseId(toDatabaseId).tableId(toTableId).build());
+    if (targetedTableDto.isPresent()) {
+      throw new AlreadyExistsException("Table", targetedTableDto.get().getTableUri());
+    }
+
+    if (isTableLocked(tableDto.get())) {
+      throw new UnsupportedClientOperationException(
+          UnsupportedClientOperationException.Operation.LOCKED_TABLE_OPERATION,
+          String.format(
+              "Table %s.%s is in locked state and cannot be updated.",
+              fromDatabaseId, fromTableId));
+    }
+
+    authorizationUtils.checkDatabasePrivilege(
```

Comment:
This should be UPDATE_TABLE_METADATA. We need to check if it is done by user who has privilege to update table.